### PR TITLE
Add recipes for Mender 2.4.2 and 2.5.1.

### PR DESCRIPTION
**Pull request by kacf**
Changelog: Add mender-artifact 3.4.1 recipe.
Changelog: Add mender-client 2.3.2 and 2.4.2 recipes.

Signed-off-by: Kristian Amlie <kristian.amlie@northern.tech>

Forward ported from dunfell, with additional tweaks to fit master:
* DISTRO_FEATURES -> MENDER_FEATURES

### CODE REVIEW

File: meta-mender-core/recipes-mender/mender-client/mender-client_2.4.2.bb [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+        install -m 755 -d ${D}/data/mender
+        install -m 444 ${B}/device_type ${D}/data/mender/
+    fi
+
+    # Setup blacklist to ensure udev does not automatically mount Mender managed partitions
+    install -d ${D}${sysconfdir}/udev/mount.blacklist.d
+    echo ${MENDER_ROOTFS_PART_A} > ${D}${sysconfdir}/udev/mount.blacklist.d/mender
+    echo ${MENDER_ROOTFS_PART_B} >> ${D}${sysconfdir}/udev/mount.blacklist.d/mender
+}
```
Comment (tranchitella) [inline meta-mender-core/recipes-mender/mender-client/mender-client_2.4.2.bb:91]: Missing new line at the end of the file

**Review by tranchitella — APPROVED** [review]

File: meta-mender-core/recipes-mender/mender-client/mender-client_2.4.2.bb (hunk 1 above)
Comment (lluiscampos) [reply meta-mender-core/recipes-mender/mender-client/mender-client_2.4.2.bb:91]: That is true. But the recipe that it is copied from had the same problem, so I would actually leave it as-is.

**Review by lluiscampos — APPROVED** [review]

File: meta-mender-core/recipes-mender/mender-client/mender-client_2.4.2.bb (hunk 1 above)
Comment (kacf) [reply meta-mender-core/recipes-mender/mender-client/mender-client_2.4.2.bb:91]: Actually, it does not! How did I manage that...

```diff
$ git show -C -C |cat
commit 77c869c5df6bbe96ced60f36741acf02431f6e29
Author: Kristian Amlie <kristian.amlie@northern.tech>
Date:   Tue Jan 26 11:15:33 2021

    Add recipes for Mender 2.4.2 and 2.5.1.
    
    Changelog: Add mender-artifact 3.4.1 recipe.
    Changelog: Add mender-client 2.3.2 and 2.4.2 recipes.
    
    Signed-off-by: Kristian Amlie <kristian.amlie@northern.tech>
    
    Forward ported from dunfell, with additional tweaks to fit master:
    * DISTRO_FEATURES -> MENDER_FEATURES

diff --git a/meta-mender-core/recipes-mender/mender-artifact/mender-artifact_3.4.0.bb b/meta-mender-core/recipes-mender/mender-artifact/mender-artifact_3.4.1.bb
similarity index 88%
copy from meta-mender-core/recipes-mender/mender-artifact/mender-artifact_3.4.0.bb
copy to meta-mender-core/recipes-mender/mender-artifact/mender-artifact_3.4.1.bb
index 815edb12..43a78c63 100644
--- a/meta-mender-core/recipes-mender/mender-artifact/mender-artifact_3.4.0.bb
+++ b/meta-mender-core/recipes-mender/mender-artifact/mender-artifact_3.4.1.bb
@@ -10,8 +10,8 @@ require mender-artifact.inc
 
 SRC_URI = "git://github.com/mendersoftware/mender-artifact.git;protocol=https;branch=3.4.x"
 
-# Tag: 3.4.0
-SRCREV = "b2c230e73b0e7c90d23ce03caedaec42728fa3f3"
+# Tag: 3.4.1
+SRCREV = "25919b61c82392653e921ec0124cde4dd7cfa405"
 
 # Enable this in Betas, not in finals.
 # Downprioritize this recipe in version selections.
@@ -24,6 +24,6 @@ SRCREV = "b2c230e73b0e7c90d23ce03caedaec42728fa3f3"
 # releases, we must check the LIC_FILES_CHKSUM.sha256 file, not the LICENSE
 # file.
 LICENSE = "Apache-2.0 & BSD-2-Clause & BSD-3-Clause & ISC & MIT"
-LIC_FILES_CHKSUM = "file://src/github.com/mendersoftware/mender-artifact/LIC_FILES_CHKSUM.sha256;md5=d1fedd6e15ea779ce58fafea700f0c37"
+LIC_FILES_CHKSUM = "file://src/github.com/mendersoftware/mender-artifact/LIC_FILES_CHKSUM.sha256;md5=7ec199eb2815c1cb147efc754c38d4b1"
 
 DEPENDS += "xz"
diff --git a/meta-mender-core/recipes-mender/mender-client/mender-client_2.3.1.bb b/meta-mender-core/recipes-mender/mender-client/mender-client_2.3.2.bb
similarity index 96%
copy from meta-mender-core/recipes-mender/mender-client/mender-client_2.3.1.bb
copy to meta-mender-core/recipes-mender/mender-client/mender-client_2.3.2.bb
index b7b4d3c3..b394c4fe 100644
--- a/meta-mender-core/recipes-mender/mender-client/mender-client_2.3.1.bb
+++ b/meta-mender-core/recipes-mender/mender-client/mender-client_2.3.2.bb
@@ -10,8 +10,8 @@ require mender-client.inc
 
 SRC_URI = "git://github.com/mendersoftware/mender;protocol=https;branch=2.3.x"
 
-# Tag: 2.3.1
-SRCREV = "c008bca66eda3993ab29d8b567bbe696e3e04866"
+# Tag: 2.3.2
+SRCREV = "4683d8708e2187b5f6b1e0b4e6da84161a15db79"
 
 # Enable this in Betas, not in finals.
 # Downprioritize this recipe in version selections.
@@ -23,7 +23,7 @@ SRCREV = "c008bca66eda3993ab29d8b567bbe696e3e04866"
 # dependencies) are included in the LICENSE variable below. Note that for
 # releases, we must check the LIC_FILES_CHKSUM.sha256 file, not the LICENSE
 # file.
-LIC_FILES_CHKSUM = "file://src/github.com/mendersoftware/mender/LIC_FILES_CHKSUM.sha256;md5=3e7426c258f60f9876ae3746f54c49d4"
+LIC_FILES_CHKSUM = "file://src/github.com/mendersoftware/mender/LIC_FILES_CHKSUM.sha256;md5=defbf977224ad6ceaf88087f40552a72"
 LICENSE = "Apache-2.0 & BSD-2-Clause & BSD-3-Clause & ISC & MIT & OLDAP-2.8"
 
 DEPENDS += "xz"
diff --git a/meta-mender-core/recipes-mender/mender-client/mender-client_2.4.1.bb b/meta-mender-core/recipes-mender/mender-client/mender-client_2.4.2.bb
similarity index 96%
copy from meta-mender-core/recipes-mender/mender-client/mender-client_2.4.1.bb
copy to meta-mender-core/recipes-mender/mender-client/mender-client_2.4.2.bb
index 00593399..d73fd5ea 100644
--- a/meta-mender-core/recipes-mender/mender-client/mender-client_2.4.1.bb
+++ b/meta-mender-core/recipes-mender/mender-client/mender-client_2.4.2.bb
@@ -10,8 +10,8 @@ require mender-client.inc
 
 SRC_URI = "git://github.com/mendersoftware/mender;protocol=https;branch=2.4.x"
 
-# Tag: 2.4.1
-SRCREV = "ffef645f932dd2fd7545705288f6e6df14417966"
+# Tag: 2.4.2
+SRCREV = "041f0b4a5df7cef49af8a80005ce57082e19be13"
 
 # Enable this in Betas, not in finals.
 # Downprioritize this recipe in version selections.
@@ -23,7 +23,7 @@ SRCREV = "ffef645f932dd2fd7545705288f6e6df14417966"
 # dependencies) are included in the LICENSE variable below. Note that for
 # releases, we must check the LIC_FILES_CHKSUM.sha256 file, not the LICENSE
 # file.
-LIC_FILES_CHKSUM = "file://src/github.com/mendersoftware/mender/LIC_FILES_CHKSUM.sha256;md5=5649992d13a6fda40abfac7730a62b07"
+LIC_FILES_CHKSUM = "file://src/github.com/mendersoftware/mender/LIC_FILES_CHKSUM.sha256;md5=d7b669e626d83679fcf25b2d82403926"
 LICENSE = "Apache-2.0 & BSD-2-Clause & BSD-3-Clause & ISC & MIT & OLDAP-2.8 & OpenSSL"
 
 DEPENDS += "xz openssl"
@@ -88,4 +88,4 @@ do_install() {
     install -d ${D}${sysconfdir}/udev/mount.blacklist.d
     echo ${MENDER_ROOTFS_PART_A} > ${D}${sysconfdir}/udev/mount.blacklist.d/mender
     echo ${MENDER_ROOTFS_PART_B} >> ${D}${sysconfdir}/udev/mount.blacklist.d/mender
-}
+}
\ No newline at end of file
```

I will fix it.

Comment (lluiscampos) [reply meta-mender-core/recipes-mender/mender-client/mender-client_2.4.2.bb:91]: Interesting. That is not what my `git` says... `2.4.1` has no newline in `dunfell`.

(the thing is that I noticed this during the original PR of these recipes for `dunfell`, then I checked that the copy was done as-is, and hence not commented on it)

**Comment by kacf** [comment]
Already passed in https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/246896322.

File: meta-mender-core/recipes-mender/mender-client/mender-client_2.4.2.bb (hunk 1 above)
Comment (kacf) [reply meta-mender-core/recipes-mender/mender-client/mender-client_2.4.2.bb]: Probably that's where it comes from then, but it just happens to be correct in master.

**Pull request merged**